ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts

```diff
@@ -509,34 +510,33 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
             tap((errorMessage) => this.stateUpdates$.next({formErrorMessages: [errorMessage]}))
         );
 
-        const removeTempAddressOperator = () => (val$: Observable<any>) => val$.pipe(
-            withLatestFrom(this.state$),
-            filter(([_, state]) => !!state.activityInstanceGuid),
-            tap(() => busyCounter$.next(1)),
-            concatMap(([_, state]) => this.addressService.deleteTempAddress(state.activityInstanceGuid)),
-            catchError(() => {
-                this.logger.logDebug(this.LOG_SOURCE, 'Temp delete failed. This is OK.');
-                return of(null);
-            }),
-            tap(() => busyCounter$.next(-1))
-        );
-
         const canSaveRealAddress = (address: Address | null) =>
             this.enoughDataToSave(address) && this.meetsActivityRequirements(address);
 
-        // "Real" as opposed to "Temp"
+        // "Real" as opposed to "Temp". Important we return the saved address as properties are added on server
         const saveRealAddressAction$ = this.saveTrigger$.pipe(
             withLatestFrom(this.formErrorMessages$, currentAddress$),
             filter(([_, formErrors, addressToSave]) => {
                 return !formErrors?.length && canSaveRealAddress(addressToSave);
             }),
             tap(() => busyCounter$.next(1)),
             concatMap(([_, formErrors, addressToSave]) => this.addressService.saveAddress(addressToSave, false)),
-            removeTempAddressOperator(),
```

Comment:
These "operator" here was the cause of the blanking of fields on submit.
We are feeding back to the form the address sent back from server after save (the server sometimes add fields values that we need to send back).
But problem here was that, at the end of the definition of the the saveRealAddressAction$ observable we were returning the result of the deletion (an HTTP response object). So no name, no street1 field etc, so corresponding form field got blanked out.
Taking the deletion out of here corrected this problem.